src/MongoDB.Driver/FieldDefinition.cs

```diff
@@ -172,13 +172,26 @@ public abstract class FieldDefinition<TDocument>
     public abstract class FieldDefinition<TDocument, TField>
     {
         /// <summary>
-        /// Renders the field to a <see cref="String"/>.
+        /// Renders the field to a <see cref="RenderedFieldDefinition{TField}"/>.
         /// </summary>
         /// <param name="documentSerializer">The document serializer.</param>
         /// <param name="serializerRegistry">The serializer registry.</param>
-        /// <returns>A <see cref="String"/>.</returns>
+        /// <returns>A <see cref="RenderedFieldDefinition{TField}"/>.</returns>
         public abstract RenderedFieldDefinition<TField> Render(IBsonSerializer<TDocument> documentSerializer, IBsonSerializerRegistry serializerRegistry);
 
+        /// Renders the field to a <see cref="RenderedFieldDefinition{TField}"/>.
+        /// <param name="documentSerializer"></param>
+        /// <param name="serializerRegistry"></param>
+        /// <param name="allowScalarValueForArrayField"></param>
+        /// <returns>A <see cref="RenderedFieldDefinition{TField}"/>.</returns>
+        public virtual RenderedFieldDefinition<TField> Render(
+            IBsonSerializer<TDocument> documentSerializer, 
+            IBsonSerializerRegistry serializerRegistry,
+            bool allowScalarValueForArrayField)
```

Comment:
I'm not really sure, but maybe it's better to add the overloading of this Render method only on levels where `allowScalarValueForArrayField` parameter makes sense? Or to have one more inheritor of FieldDefinition between FieldDefinition itself and his current children. But maybe it's too complicated.

Reply:
There is already a class hierarchy in place, so we can't have a second class hierarchy to represent when allowScalarValueForField is applicable.